### PR TITLE
Update S3 tests to Hive Metadata changes

### PR DESCRIPTION
**Pull request by findepi**
Update S3 tests code to accommodate for `SemiTransactionalHiveMetastore`
changes related to transactional tables support.

Part of https://github.com/prestosql/presto/issues/274

### CODE REVIEW

**Comment by findepi** [comment]
CI failed -- https://github.com/prestosql/presto/issues/2279

**Comment by findepi** [comment]
CI failed -- https://github.com/prestosql/presto/issues/2348, https://github.com/prestosql/presto/issues/2278

File: presto-hive-hadoop2/bin/run_hive_s3_tests.sh [hunk 1]
```diff
@@ -8,6 +8,7 @@ cleanup_docker_containers
 start_docker_containers
 
 # insert AWS credentials
+# TODO replace core-site.xml.s3-template with apply-overrides
```
Comment (electrum) [inline presto-hive-hadoop2/bin/run_hive_s3_tests.sh]: What is `apply-overrides`?

File: presto-hive-hadoop2/conf/files/hadoop-put.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -xeuo pipefail
+
+# Hadoop 3 without -d (don't create _COPYING_ temporary file) requires additional S3 permissions
+# Hadoop 2 doesn't have '-d' switch
+hadoop fs -put -f -d "$@" ||
+hadoop fs -put -f "$@"
```
[the remark this answers is not in the document]
Comment (findepi) [reply presto-hive-hadoop2/conf/files/hadoop-put.sh]: i want them on separate lines. Would you find the following more readable?

```
hadoop fs -put -f -d "$@" \
 || hadoop fs -put -f "$@"
```
?

File: presto-hive-hadoop2/bin/run_hive_s3_tests.sh (hunk 1 above)
Comment (findepi) [reply presto-hive-hadoop2/bin/run_hive_s3_tests.sh]: will update to `apply-site-xml-override.sh`

File: presto-hive-hadoop2/conf/files/hadoop-put.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (findepi) [reply presto-hive-hadoop2/conf/files/hadoop-put.sh]: i think it's actually "whatever". let's leave as is

**Comment by findepi** [comment]
Green 6 times https://github.com/prestosql/presto/runs/379179201, https://github.com/prestosql/presto/runs/379432466, https://github.com/prestosql/presto/runs/379359640
(except for unrelated flakiness which i retried on).

**Pull request merged**